decidim-core/app/controllers/decidim/account_controller.rb

```diff
@@ -56,8 +56,53 @@ def destroy
       redirect_to decidim.root_path
     end
 
+    def resend_confirmation_instructions
+      enforce_permission_to :update, :user, current_user: current_user
+
+      ResendConfirmationInstructions.call(current_user) do
+        on(:ok) do
+          respond_to do |format|
+            handle_alert(:success, t("resend_successfully", scope: "decidim.account.email_change", unconfirmed_email: current_user.unconfirmed_email))
+            format.js
+          end
+        end
+
+        on(:invalid) do
+          respond_to do |format|
+            handle_alert(:alert, t("resend_error", scope: "decidim.account.email_change"))
+            format.js
+          end
+        end
+      end
+    end
+
+    def cancel_email_change
+      enforce_permission_to :update, :user, current_user: current_user
+
+      if current_user.unconfirmed_email
+        current_user.update(unconfirmed_email: nil)
+
+        respond_to do |format|
+          handle_alert(:success, t("cancel_successfully", scope: "decidim.account.email_change"))
+          format.js
+        end
+      else
+        respond_to do |format|
+          handle_alert(:alert, t("cancel_error", scope: "decidim.account.email_change"))
+          format.js
+        end
+      end
+    end
+
     private
 
+    attr_reader :alert, :text
```

Comment:
Do you need to define `attr_reader` for these?

In the templates, I can see you are using the `@` instance variables in which case I believe you are not using the `attr_reader`s anywhere?